### PR TITLE
DBT-772: Upgrade dbt-impala adapter to use v1.6 dbt-core

### PR DESCRIPTION
**Pull request by vamshikolanu**
## Describe your changes
1. Upgrade dbt-core to 1.6
2. Removed support for microseconds datediff micro as Impala doesn't really support microseconds extration from timestamp.
3. Renamed prepare-env to dev_setup in the make file to make it consistent with dbt-hive adapter. 

## Internal Jira ticket number or external issue link
https://jira.cloudera.com/browse/DBT-772

## Testing procedure/screenshots(if appropriate):
https://gist.github.com/vamshikolanu/f6f6800ce24f638d2936addf56101b82

## Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] I have formatted my added/modified code to follow pep-8 standards
- [ ] I have checked suggestions from python linter to make sure code is of good quality.

### CODE REVIEW

File: tests/functional/adapter/test_utils.py [hunk 1]
```diff
@@ -648,7 +673,6 @@ def models(self):
 
 -- Also test correct casting of literal values.
 
-union all select {{ datediff("'1999-12-31 23:59:59.999999'", "'2000-01-01 00:00:00.000000'", "microsecond") }} as actual, 1 as expected
```
Comment (niteshy) [inline tests/functional/adapter/test_utils.py:651]: Are we removing the support for microsecond?

**Review by niteshy — COMMENTED** [review]
Few things: 
1. Let's land the separate PR for dbt-core 1.5 upgrade, so that we know test fixes are required for 1.6
2. If there are code change then let's put a separate PR so that it is easier to update the release notes. 
3. 1.6 upgrade can happen with test fixes.

File: tests/functional/adapter/test_utils.py (hunk 1 above)
Comment (vamshikolanu) [reply tests/functional/adapter/test_utils.py:651]: Yaa removed this hacky piece of code too: https://github.com/cloudera/dbt-impala/pull/196/files/5a31621c4e583939adb861501f9ca5b78fdbaa49

Comment (niteshy) [reply tests/functional/adapter/test_utils.py:651]: can we remove it as a separate commit?

**Review by niteshy — APPROVED** [review]
Please merge 1.5 PR first, then you can merge it.

**Pull request merged**